INFINITY-2998 Fetch logs using Mesos files API

nickbp opened the pull request:
- Includes executor AND agent logs
- Much faster
- Much less flaky

Sample run:
```
[2018-01-26 01:27:50,597|build.conftest|INFO]: Test test_task_dns_prefix_points_to_all_tasks failed in call phase.
[2018-01-26 01:27:50,597|build.conftest|INFO]: Fetching logs for 3 tasks launched in this suite since last failure: hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd, hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62, hello-world.d96e95ba-0237-11e8-b2f2-deead5f2b92b
[2018-01-26 01:27:50,756|sdk_cmd|INFO]: Got 200 for GET /mesos/tasks
[2018-01-26 01:27:50,935|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/debug
[2018-01-26 01:27:50,936|build.conftest|INFO]: Searching 2 paths for task Task[task_id=hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62 executor_id=hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694 agent_id=9eddb76d-3c78-4b50-8170-38d95060aa49-S4]:
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694/runs/latest
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694/runs/latest/tasks/hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62/
[2018-01-26 01:27:51,047|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/browse?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694/runs/latest
[2018-01-26 01:27:51,169|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/browse?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694/runs/latest/tasks/hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62/
[2018-01-26 01:27:51,170|build.conftest|INFO]: Downloading 4 files (15699 bytes) for task Task[task_id=hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62 executor_id=hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694 agent_id=9eddb76d-3c78-4b50-8170-38d95060aa49-S4]:
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694/runs/latest/stderr (1715 bytes)
    => logs/test_discovery_py/01__test_task_dns_prefix_points_to_all_tasks/180126_012747.hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62.executor.stderr
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694/runs/latest/stdout (485 bytes)
    => logs/test_discovery_py/01__test_task_dns_prefix_points_to_all_tasks/180126_012745.hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62.executor.stdout
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694/runs/latest/tasks/hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62/stderr (2934 bytes)
    => logs/test_discovery_py/01__test_task_dns_prefix_points_to_all_tasks/180126_012747.hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62.task.stderr
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694/runs/latest/tasks/hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62/stdout (10565 bytes)
    => logs/test_discovery_py/01__test_task_dns_prefix_points_to_all_tasks/180126_012747.hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62.task.stdout
[2018-01-26 01:27:51,280|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/download?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694/runs/latest/stderr (args: {'stream': True})
[2018-01-26 01:27:51,427|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/download?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694/runs/latest/stdout (args: {'stream': True})
[2018-01-26 01:27:51,577|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/download?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694/runs/latest/tasks/hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62/stderr (args: {'stream': True})
[2018-01-26 01:27:51,749|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/download?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__c9a0f66b-5af3-4e78-bdb4-5f43b8468694/runs/latest/tasks/hello-0-server__f39b85d0-b68e-46c3-a35c-9ff0a8114d62/stdout (args: {'stream': True})
[2018-01-26 01:27:51,750|build.conftest|INFO]: Searching 2 paths for task Task[task_id=hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd executor_id=hello__71520569-4a34-4e90-89bb-89b0565bcf2d agent_id=9eddb76d-3c78-4b50-8170-38d95060aa49-S4]:
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__71520569-4a34-4e90-89bb-89b0565bcf2d/runs/latest
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__71520569-4a34-4e90-89bb-89b0565bcf2d/runs/latest/tasks/hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd/
[2018-01-26 01:27:51,859|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/browse?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__71520569-4a34-4e90-89bb-89b0565bcf2d/runs/latest
[2018-01-26 01:27:52,002|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/browse?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__71520569-4a34-4e90-89bb-89b0565bcf2d/runs/latest/tasks/hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd/
[2018-01-26 01:27:52,003|build.conftest|INFO]: Downloading 4 files (16042 bytes) for task Task[task_id=hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd executor_id=hello__71520569-4a34-4e90-89bb-89b0565bcf2d agent_id=9eddb76d-3c78-4b50-8170-38d95060aa49-S4]:
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__71520569-4a34-4e90-89bb-89b0565bcf2d/runs/latest/stderr (2058 bytes)
    => logs/test_discovery_py/01__test_task_dns_prefix_points_to_all_tasks/180126_012743.hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd.executor.stderr
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__71520569-4a34-4e90-89bb-89b0565bcf2d/runs/latest/stdout (485 bytes)
    => logs/test_discovery_py/01__test_task_dns_prefix_points_to_all_tasks/180126_012627.hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd.executor.stdout
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__71520569-4a34-4e90-89bb-89b0565bcf2d/runs/latest/tasks/hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd/stderr (2934 bytes)
    => logs/test_discovery_py/01__test_task_dns_prefix_points_to_all_tasks/180126_012628.hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd.task.stderr
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__71520569-4a34-4e90-89bb-89b0565bcf2d/runs/latest/tasks/hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd/stdout (10565 bytes)
    => logs/test_discovery_py/01__test_task_dns_prefix_points_to_all_tasks/180126_012628.hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd.task.stdout
[2018-01-26 01:27:52,128|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/download?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__71520569-4a34-4e90-89bb-89b0565bcf2d/runs/latest/stderr (args: {'stream': True})
[2018-01-26 01:27:52,257|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/download?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__71520569-4a34-4e90-89bb-89b0565bcf2d/runs/latest/stdout (args: {'stream': True})
[2018-01-26 01:27:52,411|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/download?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__71520569-4a34-4e90-89bb-89b0565bcf2d/runs/latest/tasks/hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd/stderr (args: {'stream': True})
[2018-01-26 01:27:52,572|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/download?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0007/executors/hello__71520569-4a34-4e90-89bb-89b0565bcf2d/runs/latest/tasks/hello-0-once__89bee575-fb8e-4930-9c36-ab150c394bfd/stdout (args: {'stream': True})
[2018-01-26 01:27:52,573|build.conftest|INFO]: Downloaded 31741 bytes of logs from 2 tasks on agent 9eddb76d-3c78-4b50-8170-38d95060aa49-S4
[2018-01-26 01:27:52,689|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S4/files/download?path=/slave/log (args: {'stream': True})
[2018-01-26 01:27:54,065|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S3/files/debug
[2018-01-26 01:27:54,067|build.conftest|INFO]: Searching 1 paths for task Task[task_id=hello-world.d96e95ba-0237-11e8-b2f2-deead5f2b92b executor_id= agent_id=9eddb76d-3c78-4b50-8170-38d95060aa49-S3]:
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0001/executors/hello-world.d96e95ba-0237-11e8-b2f2-deead5f2b92b/runs/latest
[2018-01-26 01:27:54,208|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S3/files/browse?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0001/executors/hello-world.d96e95ba-0237-11e8-b2f2-deead5f2b92b/runs/latest
[2018-01-26 01:27:54,209|build.conftest|INFO]: Downloading 2 files (296876 bytes) for task Task[task_id=hello-world.d96e95ba-0237-11e8-b2f2-deead5f2b92b executor_id= agent_id=9eddb76d-3c78-4b50-8170-38d95060aa49-S3]:
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0001/executors/hello-world.d96e95ba-0237-11e8-b2f2-deead5f2b92b/runs/latest/stderr (24755 bytes)
    => logs/test_discovery_py/01__test_task_dns_prefix_points_to_all_tasks/180126_012745.hello-world.d96e95ba-0237-11e8-b2f2-deead5f2b92b.stderr
  /frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0001/executors/hello-world.d96e95ba-0237-11e8-b2f2-deead5f2b92b/runs/latest/stdout (272121 bytes)
    => logs/test_discovery_py/01__test_task_dns_prefix_points_to_all_tasks/180126_012750.hello-world.d96e95ba-0237-11e8-b2f2-deead5f2b92b.stdout
[2018-01-26 01:27:54,332|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S3/files/download?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0001/executors/hello-world.d96e95ba-0237-11e8-b2f2-deead5f2b92b/runs/latest/stderr (args: {'stream': True})
[2018-01-26 01:27:54,469|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S3/files/download?path=/frameworks/9eddb76d-3c78-4b50-8170-38d95060aa49-0001/executors/hello-world.d96e95ba-0237-11e8-b2f2-deead5f2b92b/runs/latest/stdout (args: {'stream': True})
[2018-01-26 01:27:54,581|build.conftest|INFO]: Downloaded 296876 bytes of logs from 1 tasks on agent 9eddb76d-3c78-4b50-8170-38d95060aa49-S3
[2018-01-26 01:27:54,691|sdk_cmd|INFO]: Got 200 for GET /slave/9eddb76d-3c78-4b50-8170-38d95060aa49-S3/files/download?path=/slave/log (args: {'stream': True})
```

gabrielhartmann commented:
We should give this a try on 1.9 to make sure we can still get logs from CustomExecutors

nickbp commented:
- Moved _all_ the diagnostics fetching stuff into `sdk_diag.py`, making `conftest.py` much smaller. Should make it easier for external services to use this stuff as well.
- Updated/verified to support file paths on 1.9/1.10, incl custom executors on 1.9. Needed some fallback logic but nothing too bad.
